parlai/agents/seq2seq/modules.py

```diff
@@ -27,277 +44,291 @@ def pad(tensor, length, dim=0):
-                    else:  # GRU
-                        hidden.data.copy_(new_hidden.data.index_select(dim=1, index=permute_hidden_idx))
-
-                for b in beams:
-                    b.check_finished()
-                beam_pred = [b.get_pretty_hypothesis(b.get_top_hyp()[0])[1:] for b in beams]
-                # these beam scores are rescored with length penalty!
-                beam_scores = torch.stack([b.get_top_hyp()[1] for b in beams])
-                pad_length = max([t.size(0) for t in beam_pred])
-                beam_pred = torch.stack([pad(t, length=pad_length, dim=0) for t in beam_pred], dim=0)
-
-                #  prepare n best list for each beam
-                n_best_beam_tails = [b.get_rescored_finished(n_best=len(b.finished)) for b in beams]
-                nbest_beam_scores = []
-                nbest_beam_preds = []
-                for i, beamtails in enumerate(n_best_beam_tails):
-                    perbeam_preds = []
-                    perbeam_scores = []
-                    for tail in beamtails:
-                        perbeam_preds.append(beams[i].get_pretty_hypothesis(beams[i].get_hyp_from_finished(tail)))
-                        perbeam_scores.append(tail.score)
-                    nbest_beam_scores.append(perbeam_scores)
-                    nbest_beam_preds.append(perbeam_preds)
-
-                if self.beam_log_freq > 0.0:
-                    num_dump = round(bsz * self.beam_log_freq)
-                    for i in range(num_dump):
-                        dot_graph = beams[i].get_beam_dot(dictionary=self.dict)
-                        dot_graph.write_png(os.path.join(self.beam_dump_path, "{}.png".format(self.beam_dump_filecnt)))
-                        self.beam_dump_filecnt += 1
-
-                predictions = beam_pred
-                scores = beam_scores
-
-        if isinstance(predictions, list):
-            predictions = torch.cat(predictions, 1)
-        if isinstance(scores, list):
-            scores = torch.cat(scores, 1)
-
-        return predictions, scores, cand_preds, cand_scores, encoder_states, nbest_beam_preds, nbest_beam_scores
-
-
-class Encoder(nn.Module):
-    def __init__(self, num_features, padding_idx=0, rnn_class='lstm',
-                 emb_size=128, hidden_size=128, num_layers=2, dropout=0.1,
+            scores = self._decode(encoder_states, maxlen or self.longest_label)
+
+        return scores, cand_scores, encoder_states
+
+
+class RNNEncoder(nn.Module):
+    """RNN Encoder."""
+
+    def __init__(self, num_features, embeddingsize, hiddensize,
+                 padding_idx=0, rnn_class='lstm', numlayers=2, dropout=0.1,
                  bidirectional=False, shared_lt=None, shared_rnn=None,
                  sparse=False):
+        """Initialize recurrent encoder."""
         super().__init__()
 
         self.dropout = nn.Dropout(p=dropout)
-        self.layers = num_layers
```

Comment:
Why change all the names?

Reply:
just for consistency, there was a mix of underscores and non and things didn't match with the opt args